### PR TITLE
Update requirements.txt and setup.py according to GHSA-977j-xj7q-2jr9

### PR DESCRIPTION
**Pull request by timball**
This UNTESTED patch addresses the tensorflow vulns. 

--timball

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,5 +2,5 @@ numpy~=1.14
 scipy~=1.0
 scikit-learn>=0.19
 six~=1.10
-tensorflow>=1.12.1,<2
+tensorflow>=2.0.1
```
Comment (stephen-hoover) [inline requirements.txt]: `muffnn` would require some work to switch it to use Tensorflow v2. Let's try the patched version of v1.
```suggestion
tensorflow>=1.15.2,<2
```

File: setup.py [hunk 2]
```diff
@@ -20,5 +20,5 @@
     install_requires=['numpy',
                       'scipy',
                       'scikit-learn>=0.19',
-                      'tensorflow>=1.12.1,<2']
+                      'tensorflow>=2.0.1']
```
Comment (stephen-hoover) [inline setup.py]:
```suggestion
                      'tensorflow>=1.15.2,<2']
```

File: requirements.txt (hunk 1 above)
Comment (jacksonllee) [reply requirements.txt]: +1. [The latest build run on master](https://travis-ci.org/civisanalytics/muffnn/jobs/621643563?utm_medium=notification&utm_source=github_status) did use tensorflow v1.15.0, and so I'd imagine v1.15.2 _should_ be good with muffnn as-is.

**Comment by stephen-hoover** [comment]
@timball , I don't see any additional changes. Why did you assign back to me?

Also, there's one thing I missed last time -- this will need an entry in the changelog.

**Comment by timball** [comment]
Okay @stephen-hoover sorry. didn't mean to passive-aggressive assign the ticket. I accepted your changes. Hope this works! Still totally UNTESTED.

--timball

**Comment by stephen-hoover** [comment]
@jacksonllee , it looks like Tensorflow v1.15.2 doesn't support Python 3.4 or 2.7, which this repo is still testing in. Thoughts? When will you be ready to end 2.7 and 3.4 support in `muffnn`?

The >= 3.5 tests have a couple of odd failures in the tests which I haven't had time to look into yet.

**Comment by jacksonllee** [comment]
@stephen-hoover TL;DR -- I'm leaning towards having this PR merge for now, so that this PR (and @timball ) aren't on the hook. Then the current muffnn maintainers will fix the master branch shortly (for the following two issues) and make a new release.

Re: Python 2.7 and 3.4, since Tensorflow v1.15.2 doesn't support them anymore and we need at least this version of Tensorflow for security, it looks like `muffnn` would have to stop supporting Python 2.7 and 3.4 in the next release (coming up in about a week). Probably not ideal, especially since we didn't have a cycle of deprecation, etc. But it looks like Tensorflow is the limiting factor here... The action items would be to (i) remove Python 2.7 and 3.4 in the Travis CI yaml config and (ii) document in the README.md what Python versions are supported (there's currently no mention at all).

Re: test failures for Python 3.5+, it looks like they have to do with small numerical discrepancies. I can fix them in a separate PR as well.

cc @mheilman in case you have thoughts

**Comment by jacksonllee** [comment]
@timball @stephen-hoover Thank you again for the help! @mheilman and I decided that we shouldn't merge PRs that would fail the builds, and so we're holding off this PR while investigating #102.

**Pull request merged**